packages/gatsby-link/src/index.tsx

```diff
@@ -88,9 +161,9 @@ class GatsbyLink extends React.Component {
     instance.disconnect()
   }
 
-  handleRef(ref) {
-    if (this.props.innerRef && this.props.innerRef.hasOwnProperty(`current`)) {
-      this.props.innerRef.current = ref
+  handleRef: React.Ref<HTMLAnchorElement> = (ref: HTMLAnchorElement) => {
+    if (refIsObject(this.props.innerRef)) {
```

Comment:
I don't see the benefit of extracting this as it's not reused. Can we keep the condition explicit here?

Reply:
`React.Ref<T>` is defined roughly as `React.RefFunction<T> | React.RefObject<T>`, but the check above doesn't help typescript narrow to `RefObject` in this branch  (or a function in the else branch). So the benefit of factoring out is to explicitly annotate the type assertion return type.

Alternatively, I can leave it as is and add a cast on both `if` and `else` branches.